src/main/java/io/fabric8/elasticsearch/plugin/acl/ACLDocumentManager.java

```diff
@@ -0,0 +1,283 @@
+    }
+
+    class SyncFromContextOperation implements ACLDocumentOperation {
+
+        private OpenshiftRequestContext context;
+
+        public SyncFromContextOperation(OpenshiftRequestContext context) {
+            this.context = context;
+        }
+        
+        @Override
+        public void syncDocuments(Collection<SearchGuardACLDocument> docs) {
+            LOGGER.debug("Syncing from context to ACL...");
+            for (SearchGuardACLDocument doc : docs) {
+                if(ConfigurationSettings.SEARCHGUARD_MAPPING_TYPE.equals(doc.getType())){
+                    RolesMappingSyncStrategy rolesMappingSync = documentFactory.createRolesMappingSyncStrategy((SearchGuardRolesMapping) doc);
+                    rolesMappingSync.syncFrom(context);
+                } else if(ConfigurationSettings.SEARCHGUARD_ROLE_TYPE.equals(doc.getType())) {
+                    RolesSyncStrategy rolesSync = documentFactory.createRolesSyncStrategy((SearchGuardRoles) doc);
+                    rolesSync.syncFrom(context);
+                }
+            }
+        }
+
+        @Override
+        public BulkRequest buildRequest(Client client, BulkRequestBuilder builder, Collection<SearchGuardACLDocument> docs) throws IOException{
+
+            for (SearchGuardACLDocument doc : docs) {
+                UpdateRequestBuilder update = client
+                        .prepareUpdate(searchGuardIndex, doc.getType(), SEARCHGUARD_CONFIG_ID)
+                        .setConsistencyLevel(WriteConsistencyLevel.DEFAULT)
+                        .setDoc(doc.toXContentBuilder());
+                if(doc.getVersion() != null) {
+                    update.setVersion(doc.getVersion());
+                }
+                builder.add(update.request());
+                if(LOGGER.isDebugEnabled()) {
+                    LOGGER.debug("Built {} update request: {}", doc.getType(), XContentHelper.convertToJson(doc.toXContentBuilder().bytes(),true, true));
+                }
+            }
+            return builder.request();
+        }
+        
+    }
+    
+    class ExpireOperation implements ACLDocumentOperation {
+
+        private long now;
+
+        public ExpireOperation(long currentTimeMillis) {
+            this.now = currentTimeMillis;
+        }
+
+        @Override
+        public void syncDocuments(Collection<SearchGuardACLDocument> docs) {
+            LOGGER.debug("Expiring ACLs older then {}", now);
+            for (SearchGuardACLDocument doc : docs) {
+                if(ConfigurationSettings.SEARCHGUARD_MAPPING_TYPE.equals(doc.getType())){
+                    SearchGuardRolesMapping mappings = (SearchGuardRolesMapping) doc;
+                    for (RolesMapping mapping : mappings) {
+                        //assume if the value is there its intentional
+                        if(mapping.getExpire() != null && mapping.getExpire().longValue() < now) {
```

Comment:
does `.getExpire()` return the time when this will expire? if so, should this be `>=` ?

Reply:
getExpire returns a java.util.Long being the millis since epoch.  It will only remove the mappings when the check time 'now' is 'older' then the expire time which should be '<'